<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DMAS-qPCR data analysis • DoubleqpcR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="DMAS-qPCR data analysis">
<meta property="og:description" content="DMAS qPCR analysis functions">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">DoubleqpcR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="vignette.html">Example</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AGSeifert/DoubleqpcR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="doubleqpcr">DoubleqpcR<a class="anchor" aria-label="anchor" href="#doubleqpcr"></a>
</h1></div>
<p><img align="right" width="200" src="https://user-images.githubusercontent.com/73955527/166162800-318a29a1-7e39-42d6-bad1-ef0dc28c3332.png"></p>
<p>R Package accomodating research for DMAS-qPCR experiments.</p>
<p>The default branch <code>v1.x</code> will include the most recent release.</p>
<p><strong>NEWS</strong> - 20.06.2023: <strong>v1.0.4 Paper Release</strong> - 09.11.2022: <strong>v1.0.3</strong> - 25.08.2022: <strong>v1.0.2</strong> - 11.05.2022: <strong>v1.0.1</strong> - 04.05.2022: <strong>v1.0.0 Initial Release</strong></p>
<div class="section level2">
<h2 id="about">About<a class="anchor" aria-label="anchor" href="#about"></a>
</h2>
<p>This package is designed for DMAS-qPCR data. The functions provided are useful for analysis and can handle .csv input in a specific format (see below). RDML format is not supported. This package provides the methods and functions used in the study: &gt; Detection of almonds (<em>Prunus dulcis</em>) adulteration by genotyping of sweet and bitter almonds with double-mismatch allele-specific qPCR (DMAS-qPCR) <a href="https://www.sciencedirect.com/science/article/pii/S0956713523002669" class="external-link uri">https://www.sciencedirect.com/science/article/pii/S0956713523002669</a></p>
<div class="figure">
<img src="https://github.com/LucasFVoges/DoubleqpcR/assets/73955527/2db5f7ff-3393-4c7d-b7e5-1ae2b5ea369c" alt=""><p class="caption">TOC5</p>
</div>
<p>Abstract: The main ingredient in marzipan is sweet almonds (Prunus dulcis var. dulcis). Bitter almonds (Prunus dulcis var. amara) are also added to create a more intense flavour. In Germany the proportion of bitter almonds in marzipan is limited to a maximum of 12%. The use of debittered bitter almonds is prohibited. However, currently no analytical technique can reliably monitor compliance with this legal requirement. In this study we used next-generation sequencing technology to sequence the plastid genomes of six sweet cultivars (Larguetta, Ferragnes, Nonpareil, Carmel, Monterey, and Marcona) and six bitter almond batches of different origin (Iran, Morocco, Kyrgyzstan, Spain, Syria, and Turkey) to locate single nucleotide variants (SNVs). Sweet and bitter almonds share the same maternal lineage, but sequence variants on the plastid genome were present in most bitter almond populations. To exploit these differences for detection of bitter almonds in processed products, we used the double-mismatch allele-specific qPCR (DMAS-qPCR) for two polymorphic loci (rpoB, rps4). We provide evidence that the rps4 variant was detectable in almond raw pastes containing (debittered) bitter almonds, showing that the DNA sequence was sufficiently stable throughout food processing. A clear distinction from sweet almonds was observed for samples that contained at least 8% of bitter almonds. Our results present a promising approach to detect adulterations with bitter almonds in marzipan.</p>
<div class="section level3">
<h3 id="package">Package<a class="anchor" aria-label="anchor" href="#package"></a>
</h3>
<p>A vignette (introduction) with the corresponding data from the publication listed above is available! The vignette is also available in inst/test dir with a <a href="https://agseifert.github.io/DoubleqpcR/vignette.html" class="external-link">formatted html</a></p>
<p>The package provides some basic workflow but does not claim to be universal or a ready to use Workflow. You may have to adapt certain functions for your own purpose.</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install with devtools, tough I recommend to clone the repository and work with the raw code if you eg. want to adjust functions or reuse them in your code.</p>
<p><code>devtools::install_github("LucasFVoges/DoubleqpcR", build_vignettes = FALSE)</code></p>
<p>And to include the vignette, which is also accessible in the inst/test folder from here.</p>
<p><code>devtools::install_github("LucasFVoges/DoubleqpcR", build_vignettes = TRUE)</code></p>
</div>
<div class="section level2">
<h2 id="idea">Idea<a class="anchor" aria-label="anchor" href="#idea"></a>
</h2>
<div class="section level3">
<h3 id="shifted-square-root-representation-of-a-dilution-series">Shifted square root representation of a dilution series<a class="anchor" aria-label="anchor" href="#shifted-square-root-representation-of-a-dilution-series"></a>
</h3>
<p>As reported for the DMAS-qPCR (Lefever et al. 2019) <a href="https://www.nature.com/articles/s41598-019-38581-z" class="external-link">Nature.com</a> the Delta Cq values for a de- and ascending line if plotted against the steady dilution. To make a regression model a continous data representation is neccesary. A plot against the exact proportion forms a curve which can be fitted by a 3rd degree polynomial.</p>
<p>To transform this into a linear reprensentation a shift to 0 is needed as well as the square root transformation. Therefore, to use the hole data in one regression we center 50 % at 0 wich will go up and down to 50 representing 0 and 100 % of target genotype.</p>
<p>To calculate the corresponding concentration from this shifted values, again a transformation is needed. This can be achieved with small constraints with the following formula:</p>
<p><img class="center" width="500" src="https://user-images.githubusercontent.com/73955527/166445391-81ffc119-57b5-48b2-9253-15d6f26a8b25.png"></p>
<p>where x is the shifted root of concentration (-sqrt(50) &lt; x &lt; sqrt(50)) and s_x is the sign of x.</p>
</div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p><img src="https://user-images.githubusercontent.com/73955527/186659805-246e8f40-6193-45a8-bbd4-2f75d015eb6b.png" alt="overview drawio"><a href="https://www.drawio.com/" class="external-link">made with draw.io</a></p>
<div class="section level3">
<h3 id="input-data">Input Data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>The input data has to be in csv format (or for downstream analysis a R dataframe). Two types are needed: - Fluorescence table: raw fluorescence qPCR curve data (optional for some parts) - Cq Table: a table with the two primer samples / concentration and optional Cq values</p>
<p>Example fluorescence data:</p>
<table class="table">
<thead><tr class="header">
<th>Cycle</th>
<th>A1</th>
<th>A2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>3.5</td>
<td>5.2</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
<tr class="odd">
<td>50</td>
<td>40.4</td>
<td>24.5</td>
</tr>
</tbody>
</table>
<p>Example Cq Table:</p>
<table class="table">
<thead><tr class="header">
<th>type</th>
<th>sample</th>
<th>well</th>
<th>“Cq value type1”</th>
<th>“Cq value type2”</th>
<th>…</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>gen A</td>
<td>100</td>
<td>A1</td>
<td>17.2</td>
<td>18.0</td>
<td>…</td>
</tr>
<tr class="even">
<td>gen B</td>
<td>0</td>
<td>B1</td>
<td>2.5</td>
<td>3.0</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>Important: the names of individual wells have to be coherent in both tables.</p>
<p>Sample is normally the experiment. It needs to be set to a number if later used in regression. !careful!</p>
</div>
<div class="section level3">
<h3 id="functions--workflow">Functions / Workflow<a class="anchor" aria-label="anchor" href="#functions--workflow"></a>
</h3>
<div class="section level4">
<h4 id="input">Input<a class="anchor" aria-label="anchor" href="#input"></a>
</h4>
<div class="section level5">
<h5 id="data-handling">Data Handling<a class="anchor" aria-label="anchor" href="#data-handling"></a>
</h5>
<ul>
<li>
<code><a href="reference/read.cqTable.html">read.cqTable()</a></code> will read the Cq table.</li>
<li>
<code><a href="reference/read.fluorescenceTable.html">read.fluorescenceTable()</a></code> will read the fluorescence data.</li>
</ul>
<blockquote>
<p>only .csv at this point.</p>
</blockquote>
</div>
<div class="section level5">
<h5 id="cq-calculation">Cq calculation<a class="anchor" aria-label="anchor" href="#cq-calculation"></a>
</h5>
<ul>
<li>
<code><a href="reference/calc.Cq.html">calc.Cq()</a></code> can calculate Cq values based on second / third derivative of an fitted drc model.</li>
<li>
<code><a href="reference/calc.Cq.qpcR.html">calc.Cq.qpcR()</a></code> is the same but with the qpcR package.</li>
</ul>
</div>
<div class="section level5">
<h5 id="qcpr-curves">qCPR curves<a class="anchor" aria-label="anchor" href="#qcpr-curves"></a>
</h5>
<ul>
<li>
<code><a href="reference/plot.curve.html">plot.curve()</a></code> will plot all or per sample Amplification curves</li>
</ul>
</div>
<div class="section level5">
<h5 id="cq-value-table">Cq Value table<a class="anchor" aria-label="anchor" href="#cq-value-table"></a>
</h5>
<ul>
<li>
<code><a href="reference/table.Cq.html">table.Cq()</a></code> gives an overview of the Cq values (outliers can be removed)</li>
</ul>
</div>
</div>
<div class="section level4">
<h4 id="make-data-object-for-further-analasys">Make Data Object for further analasys<a class="anchor" aria-label="anchor" href="#make-data-object-for-further-analasys"></a>
</h4>
<ul>
<li>
<code><a href="reference/make.Cq.data.html">make.Cq.data()</a></code> will generate the main list of dataframes (outliers can be removed)</li>
<li>
<code><a href="reference/combineSubsamples.html">combineSubsamples()</a></code> will combine e.g. 100.a, 100.b to one sample 100</li>
<li>
<code><a href="reference/Cq.data.mean.html">Cq.data.mean()</a></code> will generate a summery of the data (mean and sd)</li>
<li>
<code><a href="reference/Cq.data.df.html">Cq.data.df()</a></code> dataframe report</li>
<li>
<code><a href="reference/delta.Cq.data.html">delta.Cq.data()</a></code> returns a dataframe of Delta Cq values.</li>
</ul>
<blockquote>
<p>The connection to fluorescence data will be cut at this point.</p>
</blockquote>
</div>
<div class="section level4">
<h4 id="cq-plots">Cq plots<a class="anchor" aria-label="anchor" href="#cq-plots"></a>
</h4>
<ul>
<li><code><a href="reference/plot.Cq.html">plot.Cq()</a></code></li>
<li><code><a href="reference/plot.Cq.efficiency.html">plot.Cq.efficiency()</a></code></li>
</ul>
</div>
<div class="section level4">
<h4 id="delta-cq-plots">Delta Cq plots<a class="anchor" aria-label="anchor" href="#delta-cq-plots"></a>
</h4>
<ul>
<li>
<code><a href="reference/plot.delta.Cq.box.html">plot.delta.Cq.box()</a></code> Boxplots</li>
<li>
<code><a href="reference/plot.delta.Cq.overview.html">plot.delta.Cq.overview()</a></code> Data overview</li>
</ul>
</div>
<div class="section level4">
<h4 id="regression-analsysis">Regression Analsysis<a class="anchor" aria-label="anchor" href="#regression-analsysis"></a>
</h4>
<ul>
<li>
<code><a href="reference/regression.delta.Cq.html">regression.delta.Cq()</a></code> will perform model generation, plot and cross validation</li>
<li>the shifted square root transformation can be used for the regression analysis.</li>
<li>
<code><a href="reference/reTransform.html">reTransform()</a></code> will transform the values back to percentages / proportions.</li>
</ul>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Packages used in this project:</p>
<ul>
<li>investR - <a href="https://github.com/bgreenwell/investr" class="external-link">GitHub</a>
</li>
</ul>
<blockquote>
<p>Greenwell, B. M.; Schubert Kabban, C. M., investr: an R package for inverse estimation. R Journal <strong>2014</strong>, 6 (1), 90-100.</p>
</blockquote>
<ul>
<li>qpcr - <a href="http://www.dr-spiess.de/qpcR.html" class="external-link">Spiess.de</a>
</li>
</ul>
<blockquote>
<p>Ritz, C.; Spiess, A.-N., qpcR: an R package for sigmoidal model selection in quantitative real-time polymerase chain reaction analysis. Bioinformatics <strong>2008</strong>, 24 (13), 1549-1551.</p>
</blockquote>
<ul>
<li>drc <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0146021" class="external-link">PLOS ONE</a>
</li>
</ul>
<blockquote>
<p>Ritz C. et al., Dose-Response Analysis Using R, PLOS ONE, 10, 2015.</p>
</blockquote>
<ul>
<li>caret</li>
</ul>
<blockquote>
<p>Kuhn, M., Building Predictive Models in R Using the caret Package. Journal of Statistical Software <strong>2008</strong>, 28 (5), 1-26.</p>
</blockquote>
<ul>
<li>outliers - <a href="http://www.komsta.net/software" class="external-link">Komsta.net</a>
</li>
</ul>
<blockquote>
<p>Komsta, L., Processing data for outliers. The Newsletter of the R Project Volume 6/2, May 2006 2006, 6, 10.</p>
</blockquote>
<ul>
<li>ggplot2</li>
</ul>
<blockquote>
<p>Wickham, H., Ggplot2: Elegant graphics for data analysis. Springer-Verlag New York: 2016.</p>
</blockquote>
<ul>
<li>reshape2 <a href="http://www.jstatsoft.org/v21/i12/" class="external-link">JStatSoft</a>
</li>
</ul>
<blockquote>
<p>Wickham, H., Reshaping Data with the {reshape} Package. Journal of Statistical Software, 21, 2007.</p>
</blockquote>
<ul>
<li>scales <a href="https://github.com/r-lib/scales" class="external-link">GitHub</a>
</li>
</ul>
<blockquote>
<p>Wickham, H. and Seidel, D., scales: Scale Functions for Visualization. 2022.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="cite">Cite<a class="anchor" aria-label="anchor" href="#cite"></a>
</h2>
<p>If you want to cite this package please use the following citation:</p>
<pre><code>Nils Wax, Lucas F. Voges, Sören H. Wenck, Jana L. Herold, Stephan Seifert, Markus Fischer,
Detection of almonds (Prunus dulcis) adulteration by genotyping of sweet and bitter almonds with double-mismatch allele-specific qPCR (DMAS-qPCR),
Food Control,
Volume 152,
2023,
109866,
ISSN 0956-7135,
https://doi.org/10.1016/j.foodcont.2023.109866.</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/AGSeifert/DoubleqpcR/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing DoubleqpcR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Lucas F. Voges <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8433-8321" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Lucas F. Voges.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
